File: python/reverse_string_ii.py
class Solution:
    def reverseStr(self, s: str, k: int) -> str:
        result = ''
        index = 0
        while index < len(s):
            reverse_len = min(k, len(s) - index)
            for i in range(index + reverse_len - 1, index - 1, -1):
                result += s[i]
            index += k
            for i in range(index, index + k):
                if i >= len(s):
                    break
                result += s[i]
            index += k
        return result

File: src/ReverseStringII.java
public class ReverseStringII {
    public String reverseStr(String s, int k) {
        StringBuilder result = new StringBuilder();
        for (int index = 0 ; index < s.length() ; ) {
            for (int j = 0, reverseLength = Math.min(k, s.length() - index); j < reverseLength ; j++) {
                result.append(s.charAt(index + reverseLength - j - 1));
            }
            index += k;
            for (int j = 0 ; j < k && index + j < s.length() ; j++) {
                result.append(s.charAt(index + j));
            }
            index += k;
        }
        return result.toString();
    }
}
